streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -0,0 +1,465 @@
+        capturedForwards.add(new CapturedForward(To.all(), new KeyValue(key, value)));
+    }
+
+    @Override
+    public <FK, FV> void forward(final FK key, final FV value, final To to) {
+        //noinspection unchecked
+        capturedForwards.add(new CapturedForward(to, new KeyValue(key, value)));
+    }
+
+    @Override
+    public <FK, FV> void forward(final FK key, final FV value, final int childIndex) {
+        throw new UnsupportedOperationException(
+            "Forwarding to a child by index is deprecated. " +
+                "Please transition processors to forward using a 'To' object instead."
+        );
+    }
+
+    @Override
+    public <FK, FV> void forward(final FK key, final FV value, final String childName) {
+        //noinspection unchecked
+        capturedForwards.add(new CapturedForward(To.child(childName), new KeyValue(key, value)));
+    }
+
+    /**
+     * A method for retrieving all the forwarded data this context has observed. The returned list will not be
+     * affected by subsequent interactions with the context. The data in the list is in the same order as the calls to
+     * {@code forward(...)}.
+     *
+     * @return A list of key/value pairs that were previously passed to the context.
+     */
+    public List<CapturedForward> forwarded() {
+        final LinkedList<CapturedForward> result = new LinkedList<>();
+        result.addAll(capturedForwards);
+        return result;
+    }
+
+    /**
+     * A method for retrieving all the forwarded data this context has observed for a specific child by name.
+     * The returned list will not be affected by subsequent interactions with the context.
+     * The data in the list is in the same order as the calls to {@code forward(...)}.
+     *
+     * @param childName The child name to retrieve forwards for
+     * @return A list of key/value pairs that were previously passed to the context.
+     */
+    public List<CapturedForward> forwarded(final String childName) {
+        final LinkedList<CapturedForward> result = new LinkedList<>();
+        for (final CapturedForward capture : capturedForwards) {
+            if (capture.childName() == null || capture.childName().equals(childName)) {
+                result.add(capture);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Clears the captured forwarded data.
+     */
+    public void resetForwards() {
+        capturedForwards = new LinkedList<>();
+    }
+
+    // commits
+
+    @Override
+    public void commit() {
+        committed = true;
+    }
+
+    /**
+     * Whether {@link ProcessorContext#commit()} has been called in this context.
+     *
+     * @return {@code true} iff {@link ProcessorContext#commit()} has been called in this context since construction or reset.
+     */
+    public boolean committed() {
+        return committed;
+    }
+
+    /**
+     * Re-sets the commit capture to {@code false} (whether or not it was previously {@code true}).
```

Comment:
Nit: `Reset` (in the past, we agreed to use imperative always)

Reply:
k.